packages/liveblocks-core/src/auth-manager.ts

```diff
@@ -180,8 +177,11 @@ export function createAuthManager(
         (token.parsed.exp - token.parsed.iat) -
         BUFFER;
 
-      tokens.push(token);
-      expiryTimes.push(expiresAt);
+      // Legacy tokens should not get cached
+      if (token.parsed.k !== TokenKind.SECRET_LEGACY) {
+        tokens.push(token);
+        expiryTimes.push(expiresAt);
+      }
```

Comment:
@ofoucherot @marcbouchenoire I made a subtle change here while I was at it — see commit 5b3ec426d3e980b8a8a5b4f2b4615c44072315ca. Could you double-check that change, please?

I realized that legacy tokens would still get _written_ to the cache, but then ignored when trying to _read_ from the cache later. This is fine behavior as long as you _only_ have legacy tokens in the cache. But if you somehow would have a mix of token types in there, if a legacy token is written to the cache, it couldn't be garantueed that ID or Access tokens could still get read back from the cache, due to the way the read loop was constructed. (I removed those troublesome lines L87-89.)

Can you confirm this was indeed a potential bug and it's better to not even store those secret legacy tokens in the cache in the first place, like this commit is doing?

Reply:
A client should never get a mix of legacy and new tokens. And yes, I confirm it's better not to store the legacy tokens.